chat_example/src/main/java/io/skygear/chatexample/ApiParamsActivity.kt

```diff
@@ -0,0 +1,130 @@
+package io.skygear.chatexample
+
+import android.content.Intent
+import android.os.Bundle
+import android.support.v7.app.AppCompatActivity
+import android.support.v7.widget.LinearLayoutManager
+import android.widget.Toast
+import com.google.gson.Gson
+import io.skygear.chatexample.apitask.Utils
+import io.skygear.plugins.chat.ChatContainer
+import io.skygear.plugins.chat.GetParticipantsCallback
+import io.skygear.plugins.chat.Participant
+import io.skygear.skygear.Container
+import io.skygear.skygear.Error
+import kotlinx.android.synthetic.main.activity_api_params.*
+
+class ApiParamsActivity : AppCompatActivity() {
+
+    private var paramsAdapter: ApiParamsAdapter? = null
+    private var usersAdapter: ChatUsesAdapter? = null
+    val mSkygear: Container = Container.defaultContainer(this)
+    val mChatContainer: ChatContainer
+
+    init{
+        mChatContainer = ChatContainer.getInstance(mSkygear)
+    }
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_api_params)
+        title = fetchExtras().name
+
+        initViews()
+        initParams(fetchExtras())
+    }
+
+    private fun initViews() {
+        continue_btn.setOnClickListener { submitParams() }
+    }
+
+    private fun fetchExtras(): ApiTask {
+        return intent.getSerializableExtra(EXTRAS_KEY) as ApiTask
+    }
+
+    private fun initParams(apiTask: ApiTask) {
+        if(apiTask.loginRequired) {
+            // Pick User IDs
+            if (!Utils.isLoggedIn(this)) {
+                Toast.makeText(this, "Please login first!", Toast.LENGTH_SHORT).show()
+                finish()
+                return
+            }
+        }
+
+        if(apiTask.params.isEmpty()) {
+            val intent = Intent(this, ApiTestActivity::class.java)
+            intent.putExtra(EXTRAS_KEY, apiTask)
+            startActivity(intent)
+            finish()
+            return
+        }
+
+        initRecyclerView(apiTask)
+    }
+
+    private fun initRecyclerView(apiTask: ApiTask) {
+        recycler_view.layoutManager = LinearLayoutManager(this)
+
+        if(apiTask.params.containsKey(USER_ID_KEY)) {
```

Comment:
In `initRecyclerView`, we determine the ui by `apiTask.params.containsKey(USER_ID_KEY)`. And in `submitParams ` we check `paramsAdapter != null`. Since we only have 2 different kinds of input method now, so it still works. But if we have more different kinds of input method, it is easy to trigger bugs when modifying. To be better, maybe we could have a single method to determine the input method?